File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List> m = new HashMap<String,List>();
        for(String s: strs){
            char[] ca = s.toCharArray();
            Arrays.sort(ca);
            String key = String.valueOf(ca);
            if(!m.containsKey(key)){
                m.put(key, new ArrayList());
            }
            m.get(key).add(s);
        }
        return new ArrayList(m.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = defaultdict(list)
        for s in strs:
            sorted_str = "".join(sorted(s))
            if sorted_str in d:
                d[sorted_str].append(s)
            else:
                d[sorted_str] = [s]

        return list(d.values())
        